File: Programming-languages-and-laboratory/Java/Exercises/24_compito_java_2018_07_25/Ricerca.java
public class Ricerca extends Progetto {
    private int partner;
    private String arg;

    public Ricerca(int cod, String t, String c, String o, String a, int p, double i) {
        super(cod, t, c, o, i);
        this.partner = p;
        this.arg = a;
    }

    public String toString() {
        return "ricerca\t" + super.toString() + arg + "\t" + partner + "\t" + importo * 1000;
    }
}

File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Python/Exercises/24_compito_java 20180725/Ricerca.py
from Progetto import Progetto


class Ricerca(Progetto):
    def __init__(self, codice, tipo, titolo, nome, organizzazione, codice_argomento, numero_partner, importo):
        super().__init__(codice, tipo, titolo, nome, organizzazione, importo)
        self._codice_argomento = codice_argomento
        self._numero_partner = numero_partner

    def __str__(self):
        return super().__str__() + "\t" + self._codice_argomento + "\t" + str(self._numero_partner) + "\t-\t" + str(self._importo)
